torch/_inductor/ir.py

```diff
@@ -4680,53 +4705,110 @@ def require_stride1(cls, x):
         return cls.copy_input(x)
 
     @classmethod
-    def require_stride_order(cls, x, order, allow_padding=False):
+    def is_stride_ascending(cls, strides, sizes):
+        # ignore dimensions of size 1, they dont affect layout
+        non_1_indices = [
+            i
+            for i, dim in enumerate(sizes)
+            if V.graph.sizevars.size_hint(dim, fallback=2) != 1
+        ]
+
+        tmp_stride = [strides[i] for i in non_1_indices]
+        stride_size_hint = [V.graph.sizevars.size_hint(x) for x in tmp_stride]
+        # check if it is in ascending order
+        for i in range(len(tmp_stride) - 1):
+            if stride_size_hint[i] > stride_size_hint[i + 1]:
+                return False
+        return True
+
+    @classmethod
+    def require_strides(cls, x, order=None, exact_strides=None, allow_padding=False):
+        assert order is not None or exact_strides is not None
         if x.get_numel() == 0:  # Layout doesn't matter
             return x
-
-        # require x to have the layout as strided_ordered as order
+        if exact_strides:
+            exact_strides = [
+                s.node.expr if isinstance(s, torch.SymInt) else s for s in exact_strides
+            ]
```

Comment:
Symints should be handled in the border from fx graph -> inductor ir, so could we do this conversion in `graph.py` ?

Reply:
move it to graph.py in https://github.com/pytorch/pytorch/pull/130956/commits/c2b3532ed83271080d57f6393f94742691a27dd5